src/QsCompiler/CompilerOptimizations/Utils.fs

```diff
@@ -200,12 +255,30 @@ let rec internal isAllDiscarded = function
 | _ -> false
 
 
+/// Casts an int64 to an int, throwing an ArgumentException if outside the allowed range
```

Comment:
Both for this and the conversion function below it seems like you are not covering one edge case similar to this https://github.com/microsoft/qsharp-compiler/issues/86. :)

Reply:
Resolved this function in latest commit. The function below should be correct without any changes.